modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -351,7 +351,7 @@ Test_ProcQueryAlignmentProbe(void)
         QuerySetProcessValidator(QUERY,
                                  InfoClass,
                                  (PVOID)(ULONG_PTR)PsProcessInfoClass[InfoClass].AlignmentQUERY,
-                                 PsProcessInfoClass[InfoClass].RequiredSizeQUERY - 1,
+                                 0,
```

Comment:
This looks like a valid test case to me. Why do you remove this ?

Reply:
Because this test literally fails for whatever bizarre reason in ReactOS, albeit it works fine in W2k3. And as it turns out, the `RequiredSizeQUERY` value is not changed even though it should be subtracted. I dunno what's the likely cause of it so I just changed the value to 0 as it's still valid since we don't supply the correct size anyway and `STATUS_INFO_LENGTH_MISMATCH` is properly returned.

If you think you've a better suggestion or solution, I'd be glad to hear it. Thanks

Reply:
Well, that looks like a bug to me, and you remove the very test that points to it.

We must simply check sizes in NtQueryInformationProcess before probing input pointer.

Reply:
Maybe just relax the check for ICIF_QUERY_SIZE_VARIABLE & ICIF_SET_SIZE_VARIABLE in DefaultQueryInfoBufferCheck & DefaultSetInfoBufferCheck functions. I think this would make sense since those variable arrays must at least hold one value to have any effect.

Reply:
Your suggestion actually brings more problems than it solves, also sizes are already checked in NtQueryInformationProcess and friends already. I further checked again on why the test fails with `PsProcessInfoClass[InfoClass].RequiredSizeQUERY - 1` and it seems that it fails with STATUS_ACCESS_VIOLATION within `ProbeForWrite`, on that line respectively.
https://github.com/reactos/reactos/blob/master/ntoskrnl/ex/exintrin.c#L180

This case seems to happen only in NtQueryInformationProcess and NtQueryInformationThread respectively. I'm still trying to find a solution for this but dunno if I can keep going because it's so confusing as hell.

Reply:
Yes, because the test provides a non-zero length, which triggers the `Length != 0` path in ProbeForWrite. As the test provides an invalid buffer, this naturally fails.

The test actually shows that the buffer length gets tested *before* the buffer is probed, and that length check is skipped because of this line here: https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/probe.h

What happens if you comment out that "if" ?